[RDY] runtime/termdebug.vim: implement opening terminal vertically

gavinhoward opened the pull request:
All of the code was already there, including a necessary flag. This commit just adds a conditional call to `vnew` if the flag is set.

This change was suggested by @justinmk.

Commit:
runtime/termdebug.vim: implement opening terminal vertically

All of the code was already there, including a necessary flag. This
commit just adds a conditional call to 'vnew' if the flag is set.

justinmk commented on runtime/pack/dist/opt/termdebug/plugin/termdebug.vim:
```diff
@@ -140,7 +140,11 @@ endfunc
 func s:StartDebug_term(dict)
   " Open a terminal window without a job, to run the debugged program in.
-  execute 'new'
+  if s:vertical
```
```suggestion
  execute s:vertical ? 'vnew' : 'new' 
```

gavinhoward replied:
Done.

blueyed commented:
Thanks for this!

@justinmk 
Shouldn't that go to Vim also / come through it?  (we can of course foward it later, I was just wondering)

gavinhoward commented:
@blueyed Vim has already implemented it.

blueyed commented:
Ah, I see.. it does it different though: https://github.com/vim/vim/blob/6bf2c6264b5ebbe4981751840c5a8b69da08e744/runtime/pack/dist/opt/termdebug/plugin/termdebug.vim#L269-L275

Can/should we merge the current Vim version instead maybe?

blueyed commented:
Ah, but also there's this diff intentionally it seems: https://github.com/neovim/neovim/blob/84c1d3aedeb882f07bd7ff029d4ce91c50bdb9eb/runtime/pack/dist/opt/termdebug/plugin/termdebug.vim#L49-L50

blueyed commented:
btw: AppVeyor build failure is unrelated.

Commit:
runtime/termdebug.vim: add code from Vim

This code makes sure that the vertical split behaves correctly when
NeoVim is resized.

Commit:
runtime/termdebug.vim: fix a messed up character

Commit:
runtime/termdebug.vim: fix a typo
